### PR TITLE
feat(test): add support for version lock in diff tests

### PR DESCRIPTION
**Pull request by tripodsan**
- clean up fetching of original and test content
- add fetch options for original and test
- use queue to only fetch 4 pages concurrently
- improve error handling and logging
- add support for `x-ow-version-lock` header to be passed via `$TEST_VERSION_LOCK`

- add 1 hard coded url from theblog.

### CODE REVIEW

**Comment by tripodsan** [comment]
@kptdobe one thing I don't know, if how to invoke this test from another project. eg word2md and pass the `TEST_VERSION_LOCK` parameter along.

File: test/pages/pages.test.js [hunk 1]
```diff
@@ -9,19 +9,63 @@
...
+};
+
+const testOpts = {
+  cache: 'no-cache',
+  headers: {
+    'Cache-Control': 'no-cache',
+  },
+};
+
+console.log('Using test domain:', testDomain);
```
Comment (kptdobe) [inline test/pages/pages.test.js:35]: You can remove this one, CircleCI will obfuscate it anyway..

**Review by kptdobe — APPROVED** [review]

**Comment by kptdobe** [comment]
We did something similar recently for helix-publish: https://github.com/adobe/helix-publish/blob/master/.circleci/config.yml#L46-L95
It is now time to move that code into an Orb so that it is easy to re-use (remote workflow launcher and wait for result)

But you can only call a full workflow, which means your change should be applied to all tests (run_smoke_tests and run_generic_smoke_tests) so that the full test suite is executed in the context of a TEST_DOMAIN and a TEST_VERSION_LOCK.

**Pull request merged**